packages/story-editor/src/components/panels/document/taxonomies/HierarchicalTermSelector.js

```diff
@@ -203,12 +205,20 @@ function HierarchicalTermSelector({
         slug: selectedParentSlug,
       };
       createTerm(taxonomy, newCategoryName, parentValue, true);
```

Comment:
💭 can we await this so that we don't speak until the term is created? `createTerm` is an async fn so it should be gucci
```suggestion
      await createTerm(taxonomy, newCategoryName, parentValue, true);
```

Reply:
so i played with this a bit last week in terms of having it announce ahead of announcing the new focus on submit and nothing I tried got the order to change. I think it's because the live region is waiting for the next available moment to announce, given the refocus occurring AND that the new category is getting added and then announced, i think this actually makes sense. The outcome is the same regardless of awaiting `createTerm` or not, as is putting `speak` in `.then` of the `createTerm` or above the call to `createTerm`.

Reply:
> the live region is waiting for the next available moment to announce, given the refocus occurring AND that the new category is getting added and then announced

So it will always wait, even with a super slow connection? Maybe it's an edge case, but the new category could get added much later on a very slow connection.

Feels like it's better to just make sure it's queued after the `createTerm` function finishes rather than after it's called and the request gets sent, no?

Reply:
yeah i think it's tied to the toggle happening and the refocus, so those two things getting announced are more important because it's affecting the available options in the panel whereas the added category announcement isn't impacting the immediate navigability. So if we wanna keep the `add new` toggle expanded until a user toggles it manually this would probably no longer be an issue but i feel like that toggle is nice?

Reply:
I'm not sure if I follow the flow completely here, but the `await` before `createTerm` is optional for if you wanna perform an action after we get a response. just because it is a promise doesn't necessarily mean we have to do anything with the resolved value.

I would think of the promise here more like an optional callback that you can hook into if you want, but it's not necessary

Reply:
>So if we wanna keep the add new toggle expanded until a user toggles it manually this would probably no longer be an issue but i feel like that toggle is nice

I just noticed this morning with gutenberg: it keeps this UI open until the user closes it.

<img src="https://user-images.githubusercontent.com/22185279/136986607-aff6ac8b-ed7a-485f-966e-6a74d6b18939.gif" height="350px" />

> I'm not sure if I follow the flow completely here, but the `await` before `createTerm` is optional for if you wanna perform an action after we get a response. just because it is a promise doesn't necessarily mean we have to do anything with the resolved value.
> 
> I would think of the promise here more like an optional callback that you can hook into if you want, but it's not necessary

Yeah, we don't need the resolved value - but rather to wait to announce "category added" until after the category has come back from the backend

Reply:
🤔  so the reason the live region feels delayed is because the DOM is updating and that + refocusing the toggle to avoid the focus getting hijacked by the canvas is taking priority because it tells the user their immediate context and the live region is just waiting until there's a silent moment to chime in. If this becomes a concern the solution is to leave the new category form visible until the user toggles it back closed, which feels like a new issue (should that be deemed desirable), so for now I think we're good.